packages/expo-file-system/android/src/main/java/expo/modules/filesystem/unifiedfile/AssetFile.kt

```diff
@@ -0,0 +1,126 @@
+package expo.modules.filesystem.unifiedfile
+
+import android.content.Context
+import android.net.Uri
+import android.webkit.MimeTypeMap
+import androidx.core.net.toUri
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+
+class AssetFile(private val context: Context, override val uri: Uri) : UnifiedFileInterface {
+  val path: String = uri.path?.trimStart('/') ?: throw IllegalArgumentException("Invalid asset URI: $uri")
+
+  override fun exists(): Boolean = isDirectory() || isFile()
+
+  override fun isDirectory(): Boolean {
+    return context.assets.list(path)?.isNotEmpty() == true
+  }
+
+  override fun isFile(): Boolean {
+    return try {
+      context.assets.open(path).use {
+        true
+      }
+    } catch (_: Exception) {
+      false
+    }
```

Comment:
```suggestion
    return runCatching { 
      context.assets.open(path).use { true }
    }.getOrElse {
      false
    }
```

feel free to ignore. Can also be used in `length` fn below